commands/deals.go

```diff
@@ -1,24 +1,97 @@
 package commands
 
 import (
+	"encoding/json"
 	"io"
 	"strconv"
 
 	"github.com/ipfs/go-cid"
 	"github.com/ipfs/go-ipfs-cmdkit"
 	"github.com/ipfs/go-ipfs-cmds"
 	"github.com/pkg/errors"
+
+	"github.com/filecoin-project/go-filecoin/address"
+	"github.com/filecoin-project/go-filecoin/protocol/storage/storagedeal"
+)
+
+const (
+	clientOnly = "client"
+	minerOnly  = "miner"
 )
 
 var dealsCmd = &cmds.Command{
 	Helptext: cmdkit.HelpText{
 		Tagline: "Manage and inspect deals made by or with this node",
 	},
 	Subcommands: map[string]*cmds.Command{
+		"list":   dealsListCmd,
 		"redeem": dealsRedeemCmd,
 	},
 }
 
+// DealsListResult represents the subset of deal data returned by deals list
+type DealsListResult struct {
+	Miner       address.Address   `json:"minerAddress"`
+	PieceCid    cid.Cid           `json:"pieceCid"`
+	ProposalCid cid.Cid           `json:"proposalCid"`
+	State       storagedeal.State `json:"state"`
+}
+
+var dealsListCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "List all deals",
+		ShortDescription: `
+Lists all recorded deals made by or with this node. This may include pending
+deals, active deals, finished deals and cancelled deals.
+`,
+	},
+	Options: []cmdkit.Option{
+		cmdkit.BoolOption(clientOnly, "c", "only return deals made as a client"),
+		cmdkit.BoolOption(minerOnly, "m", "only return deals made as a miner"),
+	},
+	Run: func(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment) error {
+		minerAddress, _ := GetPorcelainAPI(env).ConfigGet("mining.minerAddress")
+		dealsCh, err := GetPorcelainAPI(env).DealsLs(req.Context)
+		if err != nil {
+			return err
+		}
+
+		filterForMiner, _ := req.Options[minerOnly].(bool)
+		filterForClient, _ := req.Options[clientOnly].(bool)
+
```

Comment:
does it break anything if you provide both "c" and "m"?

Reply:
it doesn't error, but it returns no deals